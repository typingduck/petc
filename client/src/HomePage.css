
.petc-homepage {
  left: 50px;
  position: absolute;
  top: 50px;
}
