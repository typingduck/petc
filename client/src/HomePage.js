import React from 'react'

import './HomePage.css'
import createNewDoc from './model/model.js'

/**
 * Default home page. Shown before user starts editing document.
 */
export default class HomePage extends React.Component {
  constructor (props) {
    super(props)

    // This binding is necessary to make `this` work in the callback
    this.handleClick = this.handleClick.bind(this)
  }

  handleClick () {
    console.log('create new')
    const doc = createNewDoc()
    console.log('create new: ' + JSON.stringify(doc))
  }

  render () {
    return (
      <div className='petc-homepage'>
        <a href='#new' onClick={this.handleClick}>Create New</a>
      </div>
    )
  }
}
