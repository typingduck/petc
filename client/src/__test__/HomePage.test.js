/* global expect it */
import React from 'react'
import Renderer from 'react-test-renderer'

import HomePage from '../HomePage'

it('matches snapshot', () => {
  const component = Renderer.create(<HomePage />)
  expect(component.toJSON()).toMatchSnapshot()
})
