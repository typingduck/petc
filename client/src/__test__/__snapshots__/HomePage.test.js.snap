// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="petc-pageview"
>
  <h1>
    pannus et circulos
  </h1>
  <div
    className="petc-homepage"
  >
    <a
      href="#new"
      id="petc-create-new"
      onClick={[Function]}
    >
      Create New
    </a>
  </div>
</div>
`;
